### PR TITLE
feat: cli mock testing

### PR DESCRIPTION
**Pull request by medjedovicm**
## Intent

- Mocking structure improved
- Added `add a "STATIC_MOCK_LOCATION=" param to .env`
- Added mocks in separate repo. [@sasjs/mocks](https://github.com/sasjs/mocks)
- ❗ Added Mock with JS STP: Mocking responses can be written with JS Stored Programs.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
- [ ] All CI checks are green.
- [x] Reviewer is assigned.

### CODE REVIEW

File: api/src/routes/web/sas9-web.ts [hunk 1]
```diff
@@ -2,13 +2,25 @@ import express from 'express'
...
 const sas9WebRouter = express.Router()
 const webController = new WebController()
 // Mock controller must be singleton because it keeps the states
 // for example `isLoggedIn` and potentially more in future mocks
 const controller = new MockSas9Controller()
 
+const mockPath = process.env.STATIC_MOCK_LOCATION || 'mocks'
+
+const upload = multer({
+  dest: path.join(process.cwd(), mockPath, 'sas9', 'files-recieved')
```
Comment (allanbowe) [inline api/src/routes/web/sas9-web.ts:21]: typo - `files-received`

File: api/src/app.ts [hunk 2]
```diff
@@ -57,6 +57,7 @@ export default setProcessVariables().then(async () => {
 
   app.use(express.json({ limit: '100mb' }))
   app.use(express.static(path.join(__dirname, '../public')))
+  app.use(express.urlencoded({ extended: true }))
```
Comment (sabhas) [inline api/src/app.ts:60]: this line is redundant

File: api/src/controllers/mock-sas9.ts [hunk 3]
```diff
@@ -16,60 +26,152 @@ export interface MockFileRead {
...
   private loggedIn: string | undefined
+  private mocksPath = process.env.STATIC_MOCK_LOCATION || 'mocks'
 
   @Get('/SASStoredProcess')
-  public async sasStoredProcess(): Promise<Sas9Response> {
+  public async sasStoredProcess(
+    @Request() req: express.Request
+  ): Promise<Sas9Response> {
+    let username = req.query._username?.toString() || undefined
+    let password = req.query._password?.toString() || undefined
```
Comment (sabhas) [inline api/src/controllers/mock-sas9.ts:36]: use const instead of let

File: api/src/controllers/mock-sas9.ts [hunk 4]
```diff
@@ -16,60 +26,152 @@ export interface MockFileRead {
...
+    if (username && password) this.loggedIn = req.body.username
+
     if (!this.loggedIn) {
       return {
         content: '',
         redirect: '/SASLogon/login'
       }
     }
 
+    let program = req.query._program?.toString() || undefined
```
Comment (sabhas) [inline api/src/controllers/mock-sas9.ts:47]: use const instead of let

File: api/src/controllers/mock-sas9.ts [hunk 5]
```diff
@@ -16,60 +26,152 @@ export interface MockFileRead {
...
+    let filePath: string[] = ['generic', 'sas-stored-process']
```
Comment (sabhas) [inline api/src/controllers/mock-sas9.ts:48]:
```suggestion
    const filePath: string[] = program ? program.replace('/', '').split('/') : ['generic', 'sas-stored-process']
```

File: api/src/controllers/mock-sas9.ts [hunk 6]
```diff
@@ -16,60 +26,152 @@ export interface MockFileRead {
...
   }
 
-  @Post('/SASStoredProcess/do/')
-  public async sasStoredProcessDo(
+  @Get('/SASStoredProcess/do')
+  public async sasStoredProcessDoGet(
     @Request() req: express.Request
   ): Promise<Sas9Response> {
+    let username = req.query._username?.toString() || undefined
+    let password = req.query._password?.toString() || undefined
```
Comment (sabhas) [inline api/src/controllers/mock-sas9.ts:73]: use const

File: api/src/controllers/mock-sas9.ts [hunk 7]
```diff
@@ -16,60 +26,152 @@ export interface MockFileRead {
...
     if (!this.loggedIn) {
       return {
         content: '',
         redirect: '/SASLogon/login'
       }
     }
 
-    if (this.isPublicAccount()) {
+    const program = req.query._program ?? req.body?._program
+    let filePath: string[] = ['generic', 'sas-stored-process']
```
Comment (sabhas) [inline api/src/controllers/mock-sas9.ts:85]: use const

**Review by sabhas — CHANGES_REQUESTED** [review]

**Review by sabhas — APPROVED** [review]

**Pull request merged**